scripts/build/TestImpactAnalysis/Testing/test_tiaf_unit_tests.py

```diff
@@ -0,0 +1,226 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+
+from logging import getLogger
+from typing import Counter
+from tiaf import TestImpact
+from tiaf_driver import main
+import pytest
+logging = getLogger("tiaf")
+
+
+def assert_list_content_equal(list1, list2):
+    assert Counter(list1) == Counter(list2)
+
+
+class TestTiafInitialiseStorage():
+
+    def test_create_TestImpact_no_s3_bucket_name(self, caplog, tiaf_args, config_data, mocker):
+        # given:
```

Comment:
For each `given`, `when` and `then`, can we get a real brief description of what's going on? I.e. "given {the starting conditions of the test}, when {these actions are performed}, then {expect this result}".